### PR TITLE
Fix for issue #252 RequestedAuthnContext: Comparison type.

### PR DESCRIPTION
**Pull request by ehsankhfr**
Fix for issue [#252 RequestedAuthnContext: Comparison type](https://github.com/node-saml/passport-saml/issue/252).

The "comparisonMethod" is added as a new parameter for config. Details:

 - Bug fix
 - Doc update
 - Tested

### CODE REVIEW

File: lib/passport-saml/config/schema.js [hunk 1]
```diff
@@ -1,8 +1,8 @@
-const SCHEME = {
+var SCHEME = {
```
Comment (markstos) [inline lib/passport-saml/config/schema.js]: It would be helpful if the commit message explained why it was necessary to replace `const` with `var` here.

Comment (ehsankhfr) [reply lib/passport-saml/config/schema.js]: With node ver4.0, Travis fails to pass. In reality:
`'const' Declarations in non-strict mode are not supported yet on Node v4.`

For the sake of consistency (like the other files), I prevented the use of strict mode.

Comment (ehsankhfr) [reply lib/passport-saml/config/schema.js]: Thanks for the great points! 👍 

I'd rather create a migration branch, to have things more transparent. Can we continue with the fix as it is, and consider that migration as a separate task?

File: .gitignore [hunk 2]
```diff
@@ -1,2 +1,15 @@
 node_modules/
 .tern-port
+
+# =========================
+# Package manger lock file
+# =========================
+
+package-lock.json
+yarn.lock
```
Comment (markstos) [inline .gitignore]: yarn.lock and package-lock.json *do* belong in Git-- at least one or the other. Distributing them insures that users install exactly the same set of dependencies that the maintainers tested with. This adds consistency which avoids strange bugs. So please remove this from the PR.

If you don't want them to show up as changed locally for you, you can use: https://stackoverflow.com/questions/1753070/how-do-i-configure-git-to-ignore-some-files-locally

Please use amend and rebase to remove the change rather adding a new commit to the branch.

File: README.md [hunk 3]
```diff
@@ -64,6 +64,7 @@ passport.use(new SamlStrategy(
   * `acceptedClockSkewMs`: Time in milliseconds of skew that is acceptable between client and server when checking `OnBefore` and `NotOnOrAfter` assertion condition validity timestamps.  Setting to `-1` will disable checking these conditions entirely.  Default is `0`.
   * `attributeConsumingServiceIndex`: optional `AttributeConsumingServiceIndex` attribute to add to AuthnRequest to instruct the IDP which attribute set to attach to the response ([link](http://blog.aniljohn.com/2014/01/data-minimization-front-channel-saml-attribute-requests.html))
   * `disableRequestedAuthnContext`: if truthy, do not request a specific authentication context. This is [known to help when authenticating against Active Directory](https://github.com/bergie/passport-saml/issues/226) (AD FS) servers.
+  * `comparisonMethod`: optional comparison method used to evaluate the requested context classes or statements, one of 'exact' (default), 'minimum', 'maximum', or 'better'.
```
Comment (markstos) [inline README.md]: Rename this to `requestedAuthContextComparison` to clarify what's being compared.

**Comment by markstos** [comment]
I've reviewed this and it appears to be a spec-compliant feature addition that appears correct, documented and tested. Once the comments I've left above are addressed, it's ready to merge.

File: .gitignore (hunk 2 above)
Comment (markstos) [reply .gitignore:9]: Refinement: `yarn.lock` has been removed from the project, but `package-lock.json` is intended to be commited.

File: lib/passport-saml/saml.js [hunk 4]
```diff
@@ -8,9 +8,11 @@ var querystring = require('querystring');
 var xmlbuilder = require('xmlbuilder');
 var xmlenc = require('xml-encryption');
 var xpath = xmlCrypto.xpath;
-var InMemoryCacheProvider = require('./inmemory-cache-provider.js').CacheProvider;
 var Q = require('q');
```
Comment (cjbarth) [inline lib/passport-saml/saml.js:11]: This should be rebased on `master` before code review continues.

**Review by cjbarth — CHANGES_REQUESTED** [review]
This need to be rebased off the head of `master` before a proper review can be done.

**Comment by ehsankhfr** [comment]
Thanks for your review, seems the current migration has already done the job of this PR (https://github.com/node-saml/passport-saml/blob/86781395c9c38dd75cbc98aaa562de8a95c225b5/src/passport-saml/saml.ts#L184)

As a result, I close this PR.

**Pull request closed without merging**